packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1138,6 +1138,13 @@ export class ContainerRuntime
 			this.mc.config.getNumber(maxConsecutiveReconnectsKey) ??
 			this.defaultMaxConsecutiveReconnects;
 
+		if (
+			runtimeOptions.flushMode === (FlushModeExperimental.Async as unknown as FlushMode) &&
+			context.supportedFeatures?.get("referenceSequenceNumbers") !== true
+		) {
+			throw new UsageError("Async FlushMode is not supported with this loader version.");
```

Comment:
I do not think throwing an error is the right thing here. You are pushing developers to check same feature before they create ContainerRuntime, to avoid hitting it. If (and I think it's true) all developers who want to use async flush mode would do same thing (i.e. fall back to non-async mode), then I'd rather see it encoded here (i.e. silent fallback to old behavior), vs. put more requirements on users and then chasing their code base (in 6 month) to clean up checking for this feature flag.

Reply:
I thought about this a lot and I'm on the fence. I'm not really comfortable ignoring and overriding an explicit opt-in feature. But otoh this feature is experimental and not widely used (at least in the near future). I guess it's fine to make the assumption and fallback on TurnBased, however I would at least post an error. The risk is to think that all is well and _fail silently_ with the expectation that stuff is supposed to work differently while it doesn't.